app/views/view_components/preview.html.erb

```diff
@@ -1,7 +1,11 @@
-<% if ViewComponent::Base.render_monkey_patch_enabled || Rails.version.to_f >= 6.1 %>
-  <%= render(@render_args[:component], @render_args[:args], &@render_args[:block]) %>
+<% if @render_args[:component] %>
+  <% if ViewComponent::Base.render_monkey_patch_enabled || Rails.version.to_f >= 6.1 %>
+    <%= render(@render_args[:component], @render_args[:args], &@render_args[:block]) %>
+  <% else %>
+    <%= render_component(@render_args[:component], &@render_args[:block]) %>
+  <% end %>
 <% else %>
-  <%= render_component(@render_args[:component], &@render_args[:block]) %>
+  <%= render template: @render_args[:template], locals: @render_args[:locals] || {} %>
```

Comment:
Looks like in Rails 5.x this is causing some issues in displaying the `preview_source`. My guess is that in app/views/view_components/_preview_source.html.erb:
```
<link href="<%= asset_path('prism.css', skip_pipeline: true) %>" media="screen" rel="stylesheet" type="text/css">
<div class="view-component-source-example">
  <h2>Source:</h2>
  <pre class="source">
    <% if @render_args[:component] %>
      <code class="language-ruby">
        <%= h @preview.preview_source(@example_name) %>
      </code>
    <% else %>
      <% template = @view_renderer.lookup_context.find_template(@render_args[:template]) %>
      <code class="language-<%= prism_language_name(template: template) %>">
        <%= h template.source %>
      </code>
    <% end %>
  </pre>
</div>
<script type="text/javascript" src="<%= asset_path('prism.min.js', skip_pipeline: true) %>"></script>
```

The `@view_renderer` doesn't know about the rendering when using this new way to render the template.

Thoughts?